File: app/demo/src/main/java/com/example/demo/web/dto/TableEnum.java
package com.example.demo.web.dto;

import com.fasterxml.jackson.annotation.JsonCreator;

public enum TableEnum {
    TAGS("TAGS"), RATINGS("RATINGS"), MOVIES("MOVIES"), LINKS("LINKS"), GENOME_SCORES("GENOME_SCORES"), GENOME_TAGS("GENOME_TAGS");

    private final String text;

    TableEnum(final String text) {
        this.text = text;
    }

    @Override
    public String toString() {
        return text;
    }

    public String getText(){return this.text;}

    @JsonCreator
    public static DatabaseEnum fromText(String text){
        for(DatabaseEnum d : DatabaseEnum.values()){
            if(d.getText().equals(text)){
                return d;
            }
        }
        throw new IllegalArgumentException();
    }
}

File: app/webapp/src/app/enum/table-enum.ts
export enum TableEnum {
  TAGS = "TAGS",
  MOVIES = "MOVIES",
  RATINGS = "RATINGS",
  LINKS = "LINKS",
  GENOME_TAGS = "GENOME_TAGS",
  GENOME_SCORES = "GENOME_SCORES"
}

export const getTableEnum = (value: string): TableEnum => {
  switch (value) {
    case TableEnum.TAGS.toString():
      return TableEnum.TAGS;
    case TableEnum.MOVIES.toString():
      return TableEnum.MOVIES;
    case TableEnum.RATINGS.toString():
      return TableEnum.RATINGS;
    case TableEnum.LINKS.toString():
      return TableEnum.LINKS;
    case TableEnum.GENOME_TAGS.toString():
      return TableEnum.GENOME_TAGS;
    default:
      return TableEnum.GENOME_SCORES;
  }
};
